python/semantic_kernel/agents/orchestration/orchestration_base.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+import inspect
+import json
+import logging
+import uuid
+from abc import ABC, abstractmethod
+from collections.abc import Awaitable, Callable
+from typing import Generic, Union, get_args
+
+from pydantic import Field
+from typing_extensions import TypeVar
+
+from semantic_kernel.agents.agent import Agent
+from semantic_kernel.agents.runtime.core.cancellation_token import CancellationToken
+from semantic_kernel.agents.runtime.core.core_runtime import CoreRuntime
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+DefaultTypeAlias = Union[ChatMessageContent, list[ChatMessageContent]]
+
+TIn = TypeVar("TIn", default=DefaultTypeAlias)
+TOut = TypeVar("TOut", default=DefaultTypeAlias)
+
+
+class OrchestrationResult(KernelBaseModel, Generic[TOut]):
+    """The result of an invocation of an orchestration."""
+
+    value: TOut | None = None
+    exception: BaseException | None = None
+    event: asyncio.Event = Field(default_factory=lambda: asyncio.Event())
```

Comment:
you don't need a lambda for this: 
```suggestion
    event: asyncio.Event = Field(default_factory=asyncio.Event)
```